### DISCUSSION TITLE
Calculate the Color Adjustment Lut From Before and After Images

### DISCUSSION THREAD

**Discussion by jindili** [libvips/libvips#2333] [Ideas]
I have a pair of Images, before.tif and after.tif 
which have the same identical contents, both of the same colorspace (RGB or CMYK),
after.tif is based on before.tif and do some color adjustments.

Is there a way in vips to calculate some kind of lut image from these two images ? and later use this lut image to batch processing lots of images.

Or if we could use some special designed image as before.tif, like the Hald Image:
![image](https://user-images.githubusercontent.com/73845/124108375-bf0b3900-da98-11eb-84c8-876f8855af7f.png)

http://www.quelsolaar.com/technology/clut.html

**Comment by jcupitt** [comment]
Hi @jindili,

No, there's no simple operation that will do this. 

I would look at adjustment profiles, as we discussed in another issue. libvips will need some small changes to make them work.
